py_ecc/bls/hash_to_curve.py

```diff
@@ -94,3 +98,27 @@ def clear_cofactor_G2(p: G2Uncompressed) -> G2Uncompressed:
     Ensure a point falls in the correct sub group of the curve.
     """
     return multiply_clear_cofactor_G2(p)
+
+
+def map_to_curve_G1(u: FQ) -> G2Uncompressed:
+    """
+    Map To Curve for G1
+
+    First, convert FQ1 point to a point on the 3-Isogeny curve.
+    SWU Map: https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-09#section-6.6.3
+
+    Second, map 3-Isogeny curve to BLS12-381-G2 curve.
+    3-Isogeny Map:
+    https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-09#name-11-isogeny-map-for-bls12-38
+    """
+    (x, y, z) = optimized_swu_G1(u)
+    return iso_map_G1(x, y, z)
```

Comment:
@petertdavies I'm seeing a `mypy` error here. I'm looking into it, but maybe you'll know more quickly:

```
py_ecc/bls/hash_to_curve.py:115: error: Incompatible return value type (got "Tuple[optimized_bls12_381_FQ, optimized_bls12_381_FQ, optimized_bls12_381_FQ]", expected "Tuple[optimized_bls12_381_FQ2, optimized_bls12_381_FQ2, optimized_bls12_381_FQ2]")  [return-value]
```

Reply:
Ah, maybe both `map_to_curve_G1` and `clear_cofactor_G1` should use `G1Uncompressed` instead of `G2Uncompressed`?